zerver/data_import/rocketchat.py

```diff
@@ -0,0 +1,673 @@
+    realm["zerver_defaultstream"] = []
+
+    return realm
+
+
+def process_users(
+    user_id_to_user_map: Dict[str, Dict[str, Any]],
+    realm_id: int,
+    domain_name: str,
+    user_handler: UserHandler,
+    user_id_mapper: IdMapper,
+) -> None:
+    for rc_user_id in user_id_to_user_map:
+        user_dict = user_id_to_user_map[rc_user_id]
+        user_dict["is_mirror_dummy"] = False
+
+        # RocketChat has three user types:
+        # "user": This is a regular user of the system.
+        # "bot": A special user types for bots.
+        # "unknown": This usually represents a livechat guest.
+        if user_dict["type"] != "user":
+            user_dict["is_mirror_dummy"] = True
+            emails = user_dict.get("emails")
+            if emails is None or len(emails) == 0:
+                user_dict["emails"] = [
+                    {
+                        "address": "{}-{}@{}".format(
+                            user_dict["username"], user_dict["type"], domain_name
+                        )
+                    }
+                ]
+
+        # TODO: Change this to use actual exported avatar
+        avatar_source = "G"
+        full_name = user_dict["name"]
+        id = user_id_mapper.get(rc_user_id)
+        delivery_email = user_dict["emails"][0]["address"]
+        email = user_dict["emails"][0]["address"]
+        short_name = user_dict["username"]
+        date_joined = int(timezone_now().timestamp())
+        timezone = "UTC"
+
+        role = UserProfile.ROLE_MEMBER
+        if "admin" in user_dict["roles"]:
+            role = UserProfile.ROLE_REALM_OWNER
+        elif "guest" in user_dict["roles"]:
+            role = UserProfile.ROLE_GUEST
+
+        if user_dict["is_mirror_dummy"]:
+            is_active = False
+            is_mirror_dummy = True
+        else:
+            is_active = True
+            is_mirror_dummy = False
+
+        user = build_user_profile(
+            avatar_source=avatar_source,
+            date_joined=date_joined,
+            delivery_email=delivery_email,
+            email=email,
+            full_name=full_name,
+            id=id,
+            is_active=is_active,
+            role=role,
+            is_mirror_dummy=is_mirror_dummy,
+            realm_id=realm_id,
+            short_name=short_name,
+            timezone=timezone,
+        )
+        user_handler.add_user(user)
+
+
+def convert_channel_data(
+    room_id_to_room_map: Dict[str, Dict[str, Any]],
+    team_id_to_team_map: Dict[str, Dict[str, Any]],
+    stream_id_mapper: IdMapper,
+    realm_id: int,
+) -> List[ZerverFieldsT]:
+    streams = []
+
+    for rc_room_id in room_id_to_room_map:
+        channel_dict = room_id_to_room_map[rc_room_id]
+
+        now = int(timezone_now().timestamp())
+        stream_id = stream_id_mapper.get(rc_room_id)
+        invite_only = channel_dict["t"] == "p"
+
+        stream_name = channel_dict["name"]
+        stream_desc = channel_dict.get("description", "")
```

Comment:
Do they have an equivalent of our `stream_post_policy` that we could import?  https://zulip.com/help/stream-sending-policy.

Reply:
I looked into it and yes, on Rocket.Chat we can mark channels as read-only, where only admins of the workplace and owners and moderators of the channel can post.

While importing such channels, I set the stream policy as -- Only admins and moderators can post.